### ISSUE TITLE
debian-kafka2 still fails when using SSL parameters in configuration

### ISSUE THREAD

**Issue by gpapaz** [fluent/fluentd-kubernetes-daemonset#1520]
### Describe the bug

Continuing on #1492, although the rdkafka is now able to be loaded, when ssl is part of the configuration for connecting to the Kafka, it still fails with the below example error messages:

```
2024-07-31 15:06:24 +0000 [warn]: #0 Send exception occurred: Configuration property "ssl.ca.location" not supported in this build: OpenSSL not available at build time at /fluentd/vendor/bundle/ruby/3.2.0/gems/rdkafka-0.11.1/lib/rdkafka/config.rb:226:in `block (2 levels) in native_config'
2024-07-31T15:06:24.193641737Z 2024-07-31 15:06:24 +0000 [warn]: #0 failed to flush the buffer. retry_times=0 next_retry_time=2024-07-31 15:06:25 +0000 chunk="61e8c70098c0ee43f541c57c26bbe45f" error_class=Rdkafka::Config::ConfigError error="Configuration property \"ssl.ca.location\" not supported in this build: OpenSSL not available at build time"
2024-07-31T15:06:24.193837801Z   2024-07-31 15:06:24 +0000 [warn]: #0 /fluentd/vendor/bundle/ruby/3.2.0/gems/rdkafka-0.11.1/lib/rdkafka/config.rb:226:in `block (2 levels) in native_config'
2024-07-31T15:06:24.193846071Z   2024-07-31 15:06:24 +0000 [warn]: #0 /fluentd/vendor/bundle/ruby/3.2.0/gems/rdkafka-0.11.1/lib/rdkafka/config.rb:216:in `each'
2024-07-31T15:06:24.193862561Z   2024-07-31 15:06:24 +0000 [warn]: #0 /fluentd/vendor/bundle/ruby/3.2.0/gems/rdkafka-0.11.1/lib/rdkafka/config.rb:216:in `block in native_config'
2024-07-31T15:06:24.193870861Z   2024-07-31 15:06:24 +0000 [warn]: #0 <internal:kernel>:90:in `tap'
2024-07-31T15:06:24.193877872Z   2024-07-31 15:06:24 +0000 [warn]: #0 /fluentd/vendor/bundle/ruby/3.2.0/gems/rdkafka-0.11.1/lib/rdkafka/config.rb:214:in `native_config'
2024-07-31T15:06:24.193883472Z   2024-07-31 15:06:24 +0000 [warn]: #0 /fluentd/vendor/bundle/ruby/3.2.0/gems/rdkafka-0.11.1/lib/rdkafka/config.rb:178:in `producer'
2024-07-31T15:06:24.193890072Z   2024-07-31 15:06:24 +0000 [warn]: #0 /fluentd/vendor/bundle/ruby/3.2.0/gems/fluent-plugin-kafka-0.19.2/lib/fluent/plugin/out_rdkafka2.rb:387:in `block in get_producer'
2024-07-31T15:06:24.193894492Z   2024-07-31 15:06:24 +0000 [warn]: #0 /fluentd/vendor/bundle/ruby/3.2.0/gems/fluent-plugin-kafka-0.19.2/lib/fluent/plugin/out_rdkafka2.rb:384:in `synchronize'
2024-07-31T15:06:24.193897642Z   2024-07-31 15:06:24 +0000 [warn]: #0 /fluentd/vendor/bundle/ruby/3.2.0/gems/fluent-plugin-kafka-0.19.2/lib/fluent/plugin/out_rdkafka2.rb:384:in `get_producer'
2024-07-31T15:06:24.193900422Z   2024-07-31 15:06:24 +0000 [warn]: #0 /fluentd/vendor/bundle/ruby/3.2.0/gems/fluent-plugin-kafka-0.19.2/lib/fluent/plugin/out_rdkafka2.rb:421:in `write'
2024-07-31T15:06:24.193903322Z   2024-07-31 15:06:24 +0000 [warn]: #0 /fluentd/vendor/bundle/ruby/3.2.0/gems/fluentd-1.17.0/lib/fluent/plugin/output.rb:1225:in `try_flush'
2024-07-31T15:06:24.193906122Z   2024-07-31 15:06:24 +0000 [warn]: #0 /fluentd/vendor/bundle/ruby/3.2.0/gems/fluentd-1.17.0/lib/fluent/plugin/output.rb:1538:in `flush_thread_run'
2024-07-31T15:06:24.193908842Z   2024-07-31 15:06:24 +0000 [warn]: #0 /fluentd/vendor/bundle/ruby/3.2.0/gems/fluentd-1.17.0/lib/fluent/plugin/output.rb:510:in `block (2 levels) in start'
2024-07-31T15:06:24.193929143Z   2024-07-31 15:06:24 +0000 [warn]: #0 /fluentd/vendor/bundle/ruby/3.2.0/gems/fluentd-1.17.0/lib/fluent/plugin_helper/thread.rb:78:in `block in thread_create'
``` 

It looks like openssl is not correctly installed on the image from this message **OpenSSL not available at build time**.
If the **libssl-dev** is included in the buildDeps, the problem is resolved as it can be seen below:

```
024-07-31 15:07:14 +0000 [warn]: #0 define <match fluent.**> to capture fluentd logs in top level is deprecated. Use <label @FLUENT_LOG> instead
2024-07-31T15:07:14.711048376Z 2024-07-31 15:07:14 +0000 [info]: #0 starting fluentd worker pid=14 ppid=7 worker=0
2024-07-31T15:07:14.711686415Z 2024-07-31 15:07:14 +0000 [info]: #0 listening port port=29999 bind="0.0.0.0"
2024-07-31T15:07:14.712611456Z 2024-07-31 15:07:14 +0000 [info]: #0 fluentd worker is now running worker=0
2024-07-31T15:07:14.736082866Z 2024-07-31 15:07:14 +0000 [error]: #0 rdkafka: [thrd:ssl://localhost:9092/bootstrap]: ssl://localhost:9092/bootstrap: Connect to ipv4#127.0.0.1:9092 failed: Connection refused (after 0ms in state CONNECT)

``` 
The connection error is due to the localhost url used that has no actual Kafka broker running.

### To Reproduce

Use the below configuration and start the fluentd image for Kafka2:
```
<source>
    @type forward
    port 29999
    bind 0.0.0.0
</source>
<match>
    @type rdkafka2
    brokers "localhost:9092"
    default_topic "test_topic"
    ssl_ca_cert "/certs/test.crt"
    <buffer>
        flush_mode immediate
    </buffer>
    <format>
      @type json
    </format>
</match>
``` 
A test certificate can be mounted under /certs/test.crt, for example the AWS ca.

### Expected behavior

Fluentd using rdkafka2 plugin is able to configure SSL specific attributes and start withou errors. 
Example successsful output was included in the description of the problem.

### Your Environment

```markdown
- Tag of using fluentd-kubernetes-daemonset: 
v1.17/debian-kafka2:v1.17.0-debian-kafka2-amd64-1.3
```

### Your Configuration

```apache
<source>
    @type forward
    port 29999
    bind 0.0.0.0
</source>
<match>
    @type rdkafka2
    brokers "localhost:9092"
    default_topic "test_topic"
    ssl_ca_cert "/certs/test.crt"
    <buffer>
        flush_mode immediate
    </buffer>
    <format>
      @type json
    </format>
</match>
```

### Your Error Log

```shell
2024-07-31 15:06:24 +0000 [warn]: #0 Send exception occurred: Configuration property "ssl.ca.location" not supported in this build: OpenSSL not available at build time at /fluentd/vendor/bundle/ruby/3.2.0/gems/rdkafka-0.11.1/lib/rdkafka/config.rb:226:in `block (2 levels) in native_config'
2024-07-31T15:06:24.193641737Z 2024-07-31 15:06:24 +0000 [warn]: #0 failed to flush the buffer. retry_times=0 next_retry_time=2024-07-31 15:06:25 +0000 chunk="61e8c70098c0ee43f541c57c26bbe45f" error_class=Rdkafka::Config::ConfigError error="Configuration property \"ssl.ca.location\" not supported in this build: OpenSSL not available at build time"
2024-07-31T15:06:24.193837801Z   2024-07-31 15:06:24 +0000 [warn]: #0 /fluentd/vendor/bundle/ruby/3.2.0/gems/rdkafka-0.11.1/lib/rdkafka/config.rb:226:in `block (2 levels) in native_config'
2024-07-31T15:06:24.193846071Z   2024-07-31 15:06:24 +0000 [warn]: #0 /fluentd/vendor/bundle/ruby/3.2.0/gems/rdkafka-0.11.1/lib/rdkafka/config.rb:216:in `each'
2024-07-31T15:06:24.193862561Z   2024-07-31 15:06:24 +0000 [warn]: #0 /fluentd/vendor/bundle/ruby/3.2.0/gems/rdkafka-0.11.1/lib/rdkafka/config.rb:216:in `block in native_config'
2024-07-31T15:06:24.193870861Z   2024-07-31 15:06:24 +0000 [warn]: #0 <internal:kernel>:90:in `tap'
2024-07-31T15:06:24.193877872Z   2024-07-31 15:06:24 +0000 [warn]: #0 /fluentd/vendor/bundle/ruby/3.2.0/gems/rdkafka-0.11.1/lib/rdkafka/config.rb:214:in `native_config'
2024-07-31T15:06:24.193883472Z   2024-07-31 15:06:24 +0000 [warn]: #0 /fluentd/vendor/bundle/ruby/3.2.0/gems/rdkafka-0.11.1/lib/rdkafka/config.rb:178:in `producer'
2024-07-31T15:06:24.193890072Z   2024-07-31 15:06:24 +0000 [warn]: #0 /fluentd/vendor/bundle/ruby/3.2.0/gems/fluent-plugin-kafka-0.19.2/lib/fluent/plugin/out_rdkafka2.rb:387:in `block in get_producer'
2024-07-31T15:06:24.193894492Z   2024-07-31 15:06:24 +0000 [warn]: #0 /fluentd/vendor/bundle/ruby/3.2.0/gems/fluent-plugin-kafka-0.19.2/lib/fluent/plugin/out_rdkafka2.rb:384:in `synchronize'
2024-07-31T15:06:24.193897642Z   2024-07-31 15:06:24 +0000 [warn]: #0 /fluentd/vendor/bundle/ruby/3.2.0/gems/fluent-plugin-kafka-0.19.2/lib/fluent/plugin/out_rdkafka2.rb:384:in `get_producer'
2024-07-31T15:06:24.193900422Z   2024-07-31 15:06:24 +0000 [warn]: #0 /fluentd/vendor/bundle/ruby/3.2.0/gems/fluent-plugin-kafka-0.19.2/lib/fluent/plugin/out_rdkafka2.rb:421:in `write'
2024-07-31T15:06:24.193903322Z   2024-07-31 15:06:24 +0000 [warn]: #0 /fluentd/vendor/bundle/ruby/3.2.0/gems/fluentd-1.17.0/lib/fluent/plugin/output.rb:1225:in `try_flush'
2024-07-31T15:06:24.193906122Z   2024-07-31 15:06:24 +0000 [warn]: #0 /fluentd/vendor/bundle/ruby/3.2.0/gems/fluentd-1.17.0/lib/fluent/plugin/output.rb:1538:in `flush_thread_run'
2024-07-31T15:06:24.193908842Z   2024-07-31 15:06:24 +0000 [warn]: #0 /fluentd/vendor/bundle/ruby/3.2.0/gems/fluentd-1.17.0/lib/fluent/plugin/output.rb:510:in `block (2 levels) in start'
2024-07-31T15:06:24.193929143Z   2024-07-31 15:06:24 +0000 [warn]: #0 /fluentd/vendor/bundle/ruby/3.2.0/gems/fluentd-1.17.0/lib/fluent/plugin_helper/thread.rb:78:in `block in thread_create'
```

**Comment by fveevaete** [comment]
@kenhys Unfortunately it seems #1492 has not been fully addressed. Could you please kindly merge @gpapaz PR? It seems it would finally fix the issue. Thank you so much.

**Comment by kenhys** [comment]
Shipped as v1.17.0-1.4 images.
